File: read.ts
import { readFileSync } from 'fs';
import { tableFromIPC } from 'apache-arrow';


console.log('reading js produced arrow file');
const jsarrow = readFileSync('jsarrow.arrow');
const jstable = tableFromIPC(jsarrow);

console.table(jstable.toArray().slice(0, 10));


console.log('reading py produced arrow file');
const pyarrow = readFileSync('pyarrow.arrow');
const pytable = tableFromIPC(pyarrow);

console.table(pytable.toArray().slice(0, 10));

File: read.py
import pyarrow as pa

print('Reading arrow ipc file produced by pyarrow...')
with open('pyarrow.arrow', 'rb') as f:
    with pa.ipc.open_file(f) as reader:
        df = reader.read_pandas()
        print(df.head())

print('Reading arrow ipc file produced by js arrow...')
with open('jsarrow.arrow', 'rb') as f:
    with pa.ipc.open_file(f) as reader:
        df = reader.read_pandas()
        print(df.head())
